index.html

```diff
@@ -495,6 +495,462 @@ <h2>
+        given [=ordered map=] |json:ordered map|, [=ordered map=]
+        |manifest:ordered map|, [=URL=] |manifest_url:URL|:
+      </p>
+      <ol class="algorithm">
+        <li>Let |processedFileHandlers:list| be a new [=list=].
+        </li>
+        <li>Set |manifest|["file_handlers"] to |processedFileHandlers|.
+        </li>
+        <li>If |json|["file_handlers"] doesn't [=map/exist=] or
+        |json|["file_handlers"] is not a [=list=], return.
+        </li>
+        <li>[=list/For each=] |entry:ordered map| of |json|["file_handlers"]:
+          <ol>
+            <li>Let |file_handler:ordered map| be [=process a file handler
+            item=] with |entry|, [=manifest/start_url=], [=manifest/scope=],
+            and |manifest_url|.
+            </li>
+            <li>If |file_handler| is failure, continue.
+            </li>
+            <li>[=list/Append=] |file_handler| to |processedFileHandlers|.
+            </li>
+          </ol>
+        </li>
+      </ol>
+      <p>
+        On [=installation=], a user agent SHOULD run the process to [=register
+        file handlers=].
+      </p>
+      <section data-cite="file-system-access">
+        <h2>
+          File Handler Items
+        </h2>
+        <p>
+          Each <dfn data-local-lt="file handler">file handler</dfn> represents
+          a URL in the scope of the application that can handle launches with
+          [=file types=] it accepts. It has the following members:
+        </p>
+        <ul>
+          <li>[=file handler/action=]
+          </li>
+          <li>[=file handler/name=]
+          </li>
+          <li>[=file handler/accept=]
+          </li>
+          <li>[=file handler/icons=]
+          </li>
+        </ul>
+        <p>
+          A user agent can use these members to associate the web application
+          with [=file type=] on the operating system.
+        </p>
+        <p>
+          A <dfn>file type</dfn> can be defined by a [=mime type=] and/or
+          [=file extension=]. A file belongs to a file type if the OS
+          determines it to have a mime type and/or its name ends with a certain
+          [=file extension=]. A <dfn>file extension</dfn> is a string that
+          start with a "." and only contains [=valid suffix code points=]. A
+          <dfn>valid suffix code point</dfn> is a [=code point=] that is
+          [=ASCII alphanumeric=], U+002B (+), or U+002E (.). Additionally,
+          [=file extensions=] are limited to a length of 16 code points.
+        </p>
+        <aside class="note">
+          <p>
+            These code points were chosen to support most pre-existing file
+            formats. The vast majority of file extensions are purely
+            alphanumeric, but compound extensions (such as .tar.gz) and
+            extensions such as .c++ for C++ source code are also fairly common,
+            hence the inclusion of + and . as allowed code points.
```

Comment:
Please see https://github.com/WICG/file-system-access/issues/363 and https://github.com/WICG/file-system-access/issues/322 for two other real-life corner cases.

Reply:
Also this link https://github.com/w3ctag/design-reviews/issues/580

Reply:
ah, thanks for pointing this out. I didn't realize there was a bit of copy-paste here (this spec has been through many revisions and has been a collaborative work among several contributors). I've linked to the file system access spec.